[SC-166] Opcja usunięcia aktywności na froncie

dmncp opened the pull request:
Zmieniłem koncepcję i dodałem do wiersza dwa przyciski - do edycji i do usuwania. Do tego od edycji podpiąłem modala z pr#165 a pod button do usunięcia modala z tego pr.

dmncp committed:
[SC-165] generalization of json editing component and using it to edit activity

dmncp committed:
[SC-166] delete activity modal

kwdrt commented on frontend/src/components/professor/ChapterDetails/ChapterDetails.js:
```diff
@@ -123,7 +149,11 @@ function ChapterDetails() {
                         </td>
                         <td>Pkt: {activity.points}</td>
-                      </tr>
+                        <td>
+                          <FontAwesomeIcon icon={faPenToSquare} onClick={() => startActivityEdition(activity)} />
+                          <FontAwesomeIcon icon={faTrash} onClick={() => deleteActivity(activity)} />
+                        </td>
+                      </TableRow>
```
Myślę że będzie fajniej jeśli damy to na oddzielne pola:
```suggestion
                        <td>
                          <FontAwesomeIcon icon={faPenToSquare} onClick={() => startActivityEdition(activity)} />
                        </td>
                        <td>
                          <FontAwesomeIcon icon={faTrash} onClick={() => deleteActivity(activity)} />
                        </td>
                      </TableRow>
```

kwdrt reviewed (approved):
Tylko jeden nonblock nit, i podoba mi się ta zmiana 👍

dmncp committed:
[SC-166] Merge remote-tracking branch 'origin/develop' into SC-166-opcja-usuniecia-aktywnosci-na-froncie

# Conflicts:
#	frontend/src/components/professor/ChapterDetails/ChapterDetails.js
#	frontend/src/components/professor/ChapterDetails/ChapterDetailsStyles.js

dmncp committed:
[SC-166] add new column to table
